pkg/event/controller.go

```diff
@@ -96,6 +100,12 @@ func (gen *generator) Add(infos ...Info) {
 			logger.V(3).Info("skipping event creation for resource without a name", "kind", info.Kind, "name", info.Name, "namespace", info.Namespace)
 			continue
 		}
+		for _, event := range gen.eventsApplied {
+			if info.Reason == Reason(event) {
+				logger.V(3).Info("skipping policy event applied event creation", "kind", info.Kind, "name", info.Name, "namespace", info.Namespace, "reason", info.Reason)
```

Comment:
May want to increase the log verbosity to 6. Otherwise we will have flooded logs.